File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = NULL;
        ListNode* current = NULL;
        ListNode* temp = head;
        ListNode* next = NULL;
        while(temp!= NULL)
        {
            
            current = temp;
            next = temp->next;
            current->next = prev;
            prev = temp;
            temp = next;
        }
    
        return current;
    }
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        prev = None
        temp = head
        current = head
        
        while temp != None:
            temp = temp.next
            current.next = prev
            prev = current
            current = temp
        
        return prev
        